### PR TITLE
[util] Rename vendor_hw.py

### PR DESCRIPTION
**Pull request by moidx**
* Rename vendor_hw.py to vendor.py
* Update doc references to util/vendor.py tool.
* Update references to util/vendor in all vendor lock files.

Closes issue lowrisc/opentitan#607

### CODE REVIEW

**Comment by sjgitty** [comment]
> This primarily lgtm but a) I just realized we have doc/ug/vendor and doc/rm/vendor; let's choose one; can you review them both and see what should survive? b) I like your suggestions of #607 and we should answer that.

BTW I think `doc/rm` is the right place for it to sit.

**Comment by imphil** [comment]
> I just realized we have doc/ug/vendor and doc/rm/vendor; let's choose one;

The documentation is intentionally split between a howto-style user guide and a reference guide. Since we didn't pick up on that distinction in other documentation I'm open to merging it (I find it very valuable though).

File: util/vendor.py [hunk 1]
```diff
@@ -17,15 +17,15 @@
 
 import hjson
 
-DESC = """vendor_hw, copy hardware source code from upstream into this repository"""
+DESC = """vendor, copy hardware source code from upstream into this repository"""
```
[the remark this answers is not in the document]
Comment (moidx) [reply util/vendor.py]: Done.

**Comment by moidx** [comment]
I will add an item to the silicon meeting agenda to close on the tool name and refactoring of documentation.

File: util/vendor.py [hunk 2]
```diff
@@ -17,15 +17,15 @@
...
+DESC = """vendor, copy source code from upstream into this repository"""
 
 EXCLUDE_ALWAYS = ['.git']
 
 LOCK_FILE_HEADER = """// Copyright lowRISC contributors.
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-// This file is generated by the vendor_hw script. Please do not modify it
+// This file is generated by the util/vendor script. Please do not modify it
```
Comment (asb) [inline util/vendor.py]: While we're changing this, "util/vendor.py script" is perhaps a better description?

**Comment by asb** [comment]
I like the vendor.py name, I think we should be more consistent about how to refer to this tool and `vendor.py` or `util/vendor.py` should be used whenever referring to the script, given we name it with the .py extension.

**Comment by imphil** [comment]
@moidx @sjgitty can we merge this as it is? It's a good improvement, and nobody really came up with an exceptional naming suggestion (at least from what I've heard).

**Pull request merged**